iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalConstant.java

```diff
@@ -27,5 +27,9 @@ private DockerLexicalConstant() {
   }
 
   public static final String COMMENT = "(?:" + LexicalConstant.SINGLE_LINE_COMMENT_HASH + ")";
+  public static final String STRING_LITERAL_WITH_QUOTES = "\"(?:[^\"\\\\]*+(?:\\\\[\\s\\S])?+)*+\"";
+  public static final String STRING_LITERAL_WITHOUT_QUOTES = ".+";
```

Comment:
This will match too much, e.g. new lines what is not expected here

Reply:
Mmhh you mean for the `.+` ? Normally it shouldn't, from my understanding `.` match anything except newline.
Or do you refer to the `[\\s\\S]` in `STRING_LITERAL_WITH_QUOTES` ? I must admit I stole this one somewhere (Sonar-Python ?) ^^' It seems quite legit to me at first to match Strings with `"`, but as I don't have a full understanding of docker syntax I may miss something ._.

Reply:
The build IntelliJ "Check RegExp" cheat me - there is DOTALL mode enabled where `.` matches a new line. Sorry for that.

Anyway, we have to think if we would store input `foo bar` as a single StringLiteral or two StringLiterals...
And we have to find another name because StringLiteral refers to the text in double quotes (See: `LexicalConstant.STRING_LITERAL`. In our cases, we usually have / should have a single word without quotes (but the Docker parser also accepts multiple words and multiple words surrounded with double quotes).